modules/tinymce/src/plugins/lists/test/ts/webdriver/ContentEditableFalseTest.ts

```diff
@@ -0,0 +1,110 @@
+import { Arr } from '@ephox/katamari';
+import { TinyAssertions, TinyHooks, TinySelections } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/lists/Plugin';
+
+interface ListParameters {
+  readonly title: string;
+  readonly content: string;
+  readonly startPath: number[];
+  readonly selector: string;
+}
+
+interface ListAction {
+  readonly title: string;
+  readonly keyPress: any[];
+}
+
+describe('webdriver.tinymce.plugins.lists.ContentEditableFalseTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: 'lists',
+    toolbar: 'numlist bullist',
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin ], true);
+
+  const listTypes = [ 'ol', 'ul' ];
+
+  const listContent =
+    '<li contenteditable="true">editable</li>\n' +
+    '<li>noneditable</li>\n' +
+    '<li contenteditable="true">editable</li>\n' +
+    '<li>noneditable</li>\n' +
+    '<li contenteditable="true">editable</li>\n';
+
+  const nonEditableListContents = (type: string): string =>
+    '<' + type + ' contenteditable="false">\n' +
+      listContent +
+    '</' + type + '>';
+
+  const divNestedNonEditableListContents = (type: string): string =>
+    '<div contenteditable="true">\n' +
+      '<' + type + ' contenteditable="false">\n' +
+        listContent +
+      '</' + type + '>\n' +
+    '</div>';
+
+  const nestedNonEditableListContents = (type1: string, type2: string): string =>
+    '<div contenteditable="false">\n' +
+      '<' + type1 + '>\n' +
+        '<li>one\n' +
+          '<' + type2 + ' contenteditable="false">\n' +
+            listContent +
+          '</' + type2 + '>\n' +
+        '</li>\n' +
+        '<li>two</li>\n' +
+      '</' + type1 + '>\n' +
+    '</div>';
+
+  const nonEditableList: ListParameters[] = Arr.bind(listTypes, (type: string) => [{
+    title: 'non-editable ' + type + ' list',
+    content: nonEditableListContents(type),
+    startPath: [ 1, 0, 0 ],
+    selector: 'iframe => body ' + type + ' li'
+  }]);
+
+  const divNestedNonEditableList: ListParameters[] = Arr.bind(listTypes, (type: string) => [{
+    title: 'non-editable div nested ' + type + ' list',
+    content: divNestedNonEditableListContents(type),
+    startPath: [ 0, 1, 0 ],
+    selector: 'iframe => body ' + type + ' li'
+  }]);
+
+  const nestedNonEditableList: ListParameters[] = Arr.bind(listTypes, (type1: string) =>
+    Arr.bind(listTypes, (type2: string) => [{
+      title: 'non-editable ' + type2 + ' list within editable ' + type1 + 'list',
+      content: nestedNonEditableListContents(type1, type2),
+      startPath: [ 1, 0, 0, 1, 0, 0 ],
+      selector: 'iframe => body ' + type1 + ' li ' + type2 + ' li'
+    }])
+  );
+
+  const contentCombinations: ListParameters[] = Arr.flatten([
+    nonEditableList,
+    divNestedNonEditableList,
+    nestedNonEditableList
+  ]);
+
+  const listActions: ListAction[] = [
+    // TODO include 'enter' key here
```

Comment:
Is this still valid? If so it should either be done as part of this PR or a Jira should be logged and the comment should be updated to reference the Jira.

Reply:
I was about to start a conversation about that, should I use the `isEditableList` function in `InsertBlock` to prevent list item splitting as well? I was thinking of asking other team members since the discussion last week.